faker/providers/address/__init__.py

```diff
@@ -93,8 +93,10 @@ def geo_coordinate(cls, center=None, radius=0.001):
         if not center:
             return Decimal(str(random.randint(-180000000, 180000000) / 1000000.0)).quantize(Decimal('.000001'))
         else:
+            center = float(center)
+            radius = float(radius)
             geo = random.uniform(center - radius, center + radius)
```

Comment:
No problem.

It might also be a good idea to replace the check from ```if not center:``` to ```if center is None:``` in the event that someone supplies a center of 0 (otherwise it'll get randomized).